### ISSUE TITLE
Not able to override Block \Magento\Theme\Block\Html\Topmenu

### ISSUE THREAD

**Issue by allamsettiramesh** [magento/magento2#12249]
I tried to override \Magento\Theme\Block\Html\Topmenu file  but i am't able to see out put.
di.xml file
 `<?xml version="1.0" encoding="UTF-8"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
    <preference for="Magento\Theme\Block\Html\Topmenu" type="Company\Cms\Block\Html\Topmenu" />   
</config>`

Block file

```
<?php

namespace Company\Cms\Block\Html;

class Topmenu extends \Magento\Theme\Block\Html\Topmenu
{

    protected function _getHtml(
        \Magento\Framework\Data\Tree\Node $menuTree,
        $childrenWrapClass,
        $limit,
        $colBrakes = []
    ) {
        echo "testt"; exit;
     }
  }

```
And i tried this blog also http://catgento.com/2017/03/20/magento-2-display-category-image-in-navigation-menu/

**Comment by allamsettiramesh** [comment]
Can any one please provide the solution for this

**Comment by jagritijoshi** [comment]
I have implemented in my local system and its working properly after disabling cache and remove folders under var/cache, var/page_cache, var/view_preprocessed. It will working fine.Please implemented this and let me know.
Thank You

**Comment by allamsettiramesh** [comment]
Hi, 
 
I tried with gethtml() method with below script


`

      public function getHtml($outermostClass = '', $childrenWrapClass = '', $limit = 0)
      {
          echo "exit"; exit;
          $this->_eventManager->dispatch(
              'page_block_html_topmenu_gethtml_before',
              ['menu' => $this->getMenu(), 'block' => $this, 'request' => $this->getRequest()]
          );

          $this->getMenu()->setOutermostClass($outermostClass);
          $this->getMenu()->setChildrenWrapClass($childrenWrapClass);

          $html = $this->_getHtml($this->getMenu(), $childrenWrapClass, $limit);

          $transportObject = new \Magento\Framework\DataObject(['html' => $html]);
          $this->_eventManager->dispatch(
              'page_block_html_topmenu_gethtml_after',
              ['menu' => $this->getMenu(), 'transportObject' => $transportObject]
          );
          $html = $transportObject->getHtml();
          return $html;
      }
 `
  

but still it's does't return what i printed as per below screen shot
![menu](https://user-images.githubusercontent.com/16919356/32893699-793ba294-cb00-11e7-8cc0-5f0d3d7bee41.png)

**Comment by jagritijoshi** [comment]
Hello ,
   I tried and it's working properly.I have attached screenshot below.

Thanks!!

![localhost contribute](https://user-images.githubusercontent.com/23253057/32944231-d3cc90a6-cbb4-11e7-8b1b-75fd2028c006.png)

**Comment by jagritijoshi** [comment]
Hello 
   I have attached code folder.Please upload it and then check.

thanks!!
[Company.zip](https://github.com/magento/magento2/files/1482020/Company.zip)

**Comment by magento-engcom-team** [comment]
@allamsettiramesh, thank you for your report.
The issue is already fixed in develop branch, 2.2.1
